rust/src/downloads.rs

```diff
@@ -0,0 +1,46 @@
+use std::error::Error;
+use std::fs::File;
+use std::io::copy;
+use std::io::Cursor;
+
+use tempfile::{Builder, TempDir};
+
+use crate::files::parse_version;
+
+#[tokio::main]
```

Comment:
This should just be async instead of using the tokio::main macro, this prevents it from being used in existing projects that are async, this code can also just be sync.

Reply:
Would you like to send us a pull request to improve the code?